fe/fe-core/src/main/java/com/starrocks/sql/common/MetaUtils.java

```diff
@@ -136,18 +136,46 @@ public static Table getTable(TableName tableName) {
         return table;
     }
 
+    // get the Table in the session based on table name,
+    // if there are temporary tables and normal tables with the same name, the temporary table will be used first.
     public static Table getTable(ConnectContext session, TableName tableName) {
         if (Strings.isNullOrEmpty(tableName.getCatalog())) {
             tableName.setCatalog(session.getCurrentCatalog());
         }
-        Table table = session.getGlobalStateMgr().getMetadataMgr().getTable(tableName.getCatalog(),
-                tableName.getDb(), tableName.getTbl());
+        Database database = getDatabase(session, tableName);
+
+        Table table = session.getGlobalStateMgr().getMetadataMgr().getTemporaryTable(
```

Comment:
Is it possible to merge getTemporaryTable and getTable? I think most places do not distinguish whether it is a temporary table or not.

Reply:
> Is it possible to merge getTemporaryTable and getTable? I think most places do not distinguish whether it is a temporary table or not.

I've add a new interface `getSessionAwareTable` to solve it.